### ISSUE TITLE
Visual mode surround repeat

### ISSUE THREAD

**Issue by unphased** [tpope/vim-surround#176]
I know I can use `ys`, etc but it's more immediate and has feedback if I pick out my subject using visual mode first. 

E.g. `ys2awffunc` requires me a lot more thinking to properly execute, but `v2awffunc` is both one keystroke shorter and would offer visual feedback prior to committing to it, AND it lets me correct my invariable mistake prior to committing to the operation.

The visual mode operation still works wonderfully which is ace! but vim-repeat doesn't register it. 

For visual the built-in repeat behavior is to just treat the subject as the length of the visual selection, starting from the cursor position.

Basically I have taken to use visual mode to trigger surround most of the time and i wonder if we can use the length of the last visual block to use for repeat rather than completely ignoring it. 

One distinct possibility is that the visual mode is fundamentally limited and I basically just need to use the `ys` binds to do the job. In which case, sadness.

**Comment by tommcdo** [comment]
Repeating Visual operations rarely works nicely. Since you can perform an arbitrary series of motions before your Visual selection is "complete", there's no good way for Vim to duplicate that when it repeats the operation with `.`. So it does the best it can: it operates on the same number of characters (or lines).

So if you were planning to use `.` to surround another 2 words in a function call, you probably wouldn't get the results you wanted even if surround.vim *did* integrate with repeat.vim on Visual operations.

Since repeating Visual operations is so unreliable (even with built-in operators), I question the merit of supporting it in custom operators.

**Comment by unphased** [comment]
I think you’re right to question the merit of trying to support it. 

perhaps what neovim needs is a separate feature that can report on what the actions were that were performed in the last visual mode edit. then we can use them to play them back when vim repeat is being invoked.

**Comment by tpope** [comment]
Yep.

**Comment by unphased** [comment]
Is there simply a way to make this not do anything? 

I will press `.` after doing a visual-surround from reflex, or whatever, and it does something Bad™ to my buffer when I do it.
